test/foundry/unit/concrete/erc4337account/TestERC4337Account_AddDeposit.t.sol

```diff
@@ -0,0 +1,151 @@
+        assertTrue(res, "Pre-funding account should succeed");
+
+        vm.expectEmit(true, true, true, true);
+        emit DepositAdded(address(account), address(this), depositAmount); // Assuming there's a DepositAdded event
+
+        account.addDeposit{value: depositAmount }();
+    }
+
+    function test_AddDeposit_Revert_NoValue() public {
+        account.addDeposit();
+    }
+
+function test_AddDeposit_DepositViaHandleOps() public {
+    (bool res, ) = payable(address(BOB_ACCOUNT)).call{ value: 2 ether }(""); // Fund account with 2 ETH
+    assertTrue(res, "Pre-funding account should succeed");
+    uint256 depositAmount = 1 ether;
+    uint balanceBefore = ENTRYPOINT.balanceOf(address(account));
+
+    Execution[] memory execution = new Execution[](1);
+    execution[0] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+
+    // Build UserOperation for single execution
+    PackedUserOperation[] memory userOps = prepareUserOperation(
+        BOB,
+        account,
+        EXECTYPE_DEFAULT,
+        execution
+    );
+        uint256 gasStart = gasleft();
+    ENTRYPOINT.handleOps(userOps, payable(BOB.addr));
+    uint gasUsed = gasStart - gasleft();
+
+    // Using almostEq to compare balances with a tolerance for gas costs
+    uint256 maxPercentDelta = 100_000_000_000;
+    almostEq(balanceBefore + depositAmount  - (gasUsed * tx.gasprice), ENTRYPOINT.balanceOf(address(account)), maxPercentDelta);
+}
+
+function test_AddDeposit_BatchDepositViaHandleOps() public {
+    (bool res, ) = payable(address(BOB_ACCOUNT)).call{ value: 10 ether }(""); // Fund account with 2 ETH
+    assertTrue(res, "Pre-funding account should succeed");
+    uint256 depositAmount = 1 ether;
+    uint balanceBefore = ENTRYPOINT.balanceOf(address(account));
+
+    Execution[] memory executions = new Execution[](5);
+    executions[0] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+    executions[1] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+    executions[2] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+    executions[3] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+    executions[4] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
+
+    // Build UserOperation for single execution
+    PackedUserOperation[] memory userOps = prepareUserOperation(
+        BOB,
+        account,
+        EXECTYPE_DEFAULT,
+        executions
+    );
+        uint256 gasStart = gasleft();
+    ENTRYPOINT.handleOps(userOps, payable(BOB.addr));
+    uint gasUsed = gasStart - gasleft();
+
+    // Using almostEq to compare balances with a tolerance for gas costs
+    uint256 maxPercentDelta = 100_000_000_000;
+    almostEq(balanceBefore + (depositAmount * 5)  - (gasUsed * tx.gasprice), ENTRYPOINT.balanceOf(address(account)), maxPercentDelta);
```

Comment:
can do this test with paymaster involved, then it should be equal

Reply:
Yes, can add additional tests with a MockPaymaster